pkg/proctree/proctree_procfs.go

```diff
@@ -151,32 +160,24 @@ func (pt *ProcessTree) FeedFromProcFS() error {
 		return nil
 	}
 
-	// Walk the procfs tree
-
-	dirs, err := os.ReadDir(procDir)
-	if err != nil {
-		return errfmt.Errorf("could not read proc dir: %v", err)
-	}
-	for _, dir := range dirs {
-		if !dir.IsDir() {
-			continue
-		}
-		pid, err := strconv.Atoi(dir.Name())
-		if err != nil {
-			continue
+	// dealWithProcFsEntry deals with a process from procfs.
+	dealWithProcFsEntry := func(p int) error {
```

Comment:
I don't see it as a problem, but, after going through some runtime readings I saw:

```
If the anonymous function does capture variables, there's a small overhead to create the closure, but this is often negligible in practice unless you're in a very tight loop or a performance-critical path.
```

In our case I'm using the "*ProcTree" variable as a captured variable so I was in doubt if that could cause performance issues. Because of this doubt I created:

1. func getProcessByPID(pt *ProcessTree, givenPid int) (*Process, error)
2. func dealWithProc(pt *ProcessTree, givenPid int) error
3. func dealWithThread(pt *ProcessTree, pid int, tid int) error
4. func dealWithProcFsEntry(pt *ProcessTree, givenPid int) error

And the reason for not doing them as ProcessTree internal methods is that I don't want these functions exposed through the process tree API (same reason why I kept them anonymous in the 1st place).